mkosi/config.py

```diff
@@ -4297,6 +4297,8 @@ def parse_config_one(self, path: Path, parse_profiles: bool = False, parse_local
         s: Optional[ConfigSetting[object]]  # Hint to mypy that we might assign None
         extras = path.is_dir()
 
+        assert path.is_absolute()
```

Comment:
For typing assert is fine, but we shouldn't use it for errors.
```suggestion
        if not path.is_absolute():
            raise ValueError(f'{path} is not absolute')
```

Reply:
@behrmann This is a programming error, not a user error, so an assert fits better imo